code/game/objects/items/reagent_containers/blood_pack.dm

```diff
@@ -13,22 +13,39 @@
 	if(blood_type)
 		name = "BloodPack [blood_type]"
 		reagents.add_reagent(/datum/reagent/blood, 200, list("donor"=null,"blood_DNA"=null,"blood_type"=blood_type))
-		update_icon()
-
 
 /obj/item/reagent_containers/blood/on_reagent_change()
-	update_icon()
+	update_overlays()
+
+/obj/item/reagent_containers/blood/update_overlays()
+	. = ..()
 
-/obj/item/reagent_containers/blood/update_icon_state()
+	overlays.Cut()
+	if(reagents.total_volume)
+		var/image/filling = image('icons/obj/reagentfillings.dmi', src, "[icon_state]10")
 
-	var/percent = PERCENT(reagents.total_volume / volume)
-	switch(percent)
-		if(0 to 9.9)
-			icon_state = "empty"
-		if(10 to 50)
-			icon_state = "half"
-		if(50.1 to INFINITY)
-			icon_state = "full"
+		var/percent = round((reagents.total_volume / volume) * 100)
+		switch(percent)
+			if(0 to 9)
+				filling.icon_state = "[initial(icon_state)]-10"
+			if(10 to 24)
+				filling.icon_state = "[initial(icon_state)]10"
+			if(25 to 49)
+				filling.icon_state = "[initial(icon_state)]25"
+			if(50 to 74)
+				filling.icon_state = "[initial(icon_state)]50"
+			if(75 to 79)
+				filling.icon_state = "[initial(icon_state)]75"
+			if(80 to 90)
+				filling.icon_state = "[initial(icon_state)]80"
+			if(91 to INFINITY)
+				filling.icon_state = "[initial(icon_state)]100"
+
+		filling.color = mix_color_from_reagents(reagents.reagent_list)
+		overlays += filling
+
+	if(blood_type)
+		add_overlay(image('icons/obj/items/bloodpack.dmi', "[blood_type]"))
```

Comment:
incorrect update overlays usage
The way it works is you call update_icon and then update_overlays returns a list of overlays to add